compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/test/dma_cse.mlir

```diff
@@ -0,0 +1,71 @@
+// RUN: iree-opt --pass-pipeline="builtin.module(func.func(iree-amdaie-dma-cse))" --split-input-file %s --verify-diagnostics | FileCheck %s
+
+// CHECK-LABEL: @circular_dma_perform_cse
+// CHECK:       %[[CONNECTION_1:.+]] = amdaie.connection
+// CHECK:       %[[CONNECTION_2:.+]] = amdaie.connection
+// CHECK:       amdaie.controlcode
+// CHECK:         scf.for
+// CHECK:           amdaie.npu.circular_dma_cpy_nd %[[CONNECTION_1]]([] [] [], [] [] [])
+// CHECK:           amdaie.npu.circular_dma_cpy_nd %[[CONNECTION_2]]([0] [16] [1], [] [] [])
+// CHECK:           scf.forall
+#executable_target_amdaie_xclbin_fb = #hal.executable.target<"amd-aie", "amdaie-xclbin-fb", {target_device = "npu1_4col", ukernels = "none"}>
+module attributes {hal.executable.target = #executable_target_amdaie_xclbin_fb} {
+  func.func @circular_dma_perform_cse(%arg0: !amdaie.logicalobjectfifo<memref<1x1x8x16xi32>>, %arg1: !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>) {
+    %c1 = arith.constant 1 : index
+    %c2 = arith.constant 2 : index
+    %c6 = arith.constant 6 : index
+    amdaie.workgroup {
+      %0 = amdaie.connection(%arg0, %arg1) : (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+      %1 = amdaie.connection(%arg0, %arg1) : (!amdaie.logicalobjectfifo<memref<1x1x8x16xi32>>, !amdaie.logicalobjectfifo<memref<8x16xi32, 1>>)
+      amdaie.controlcode {
+        scf.for %arg2 = %c1 to %c6 step %c2 {
+          %2 = amdaie.npu.circular_dma_cpy_nd %0([] [] [], [] [] [])
+          %3 = amdaie.npu.circular_dma_cpy_nd %0([] [] [], [] [] [])
```

Comment:
This doesn't look like cse to me. Is this like 

```
copy(dst, src)
copy(dst, src)
```

becoming 
```
copy(dst, src)
```
Because the first copy is dead? Seems more like a liveness analysis thing to me, if so. CSE is something in value semantics, no? 

Why is the scf.forall retained, is this a future optimization? i.e. %5 and %4 are the same aren't they (imagine unrolling).

Reply:
> No, because (in our case) the copy is circular so the DMAs don't need to be programmed twice.
Makes sense. 

> Will be removed if you run cse after this.

Ok good. So maybe add extra lines

// CHECK-NOT: circular_dma_cpy_nd
// CHECK: amdaie.end

Reply:
> Ok good. So maybe add extra lines
> // CHECK-NOT: circular_dma_cpy_nd
> // CHECK: amdaie.end

Done